src/app/body/milestone-details/milestone-details.component.ts

```diff
@@ -173,9 +174,24 @@ export class MilestoneDetailsComponent implements OnInit {
     var width = (100 / (this.milestoneService.taskData.length)).toString() + "%";
     return width;
   };
+
+  setMilestoneWidth2 = function () {
```

Comment:
Instead of having separate functions for milestone widths try to optimize and do it within the previous implementation